google-cloud-spanner/lib/google/cloud/spanner/results.rb

```diff
@@ -26,7 +26,12 @@ class Results
         ##
         # Indicates the field names and types for the rows in the returned data.
         #
-        # @return [Hash] The types of the returned data.
+        # @param [Boolean] pairs Allow the types to be represented as a nested
+        #   Array of pairs rather than a Hash. This is useful when results have
+        #   duplicate names. The default is `false`.
```

Comment:
If someone has duplicate columns in his query, then I guess he would want to know the value of both the columns, otherwise why did he ask for both the columns in the first place. In that case why would he want to get back a Hash which removes the values of one of the columns. I would think in most cases, asking for the Hash when you have duplicate columns is a bug and throwin an error would make it easier to realise that.